### PR TITLE
chore(java): improve memory usage

### PR DESCRIPTION
**Pull request by gastonfournier**
Make some changes in UnleashEngine internals to release memory properly

### CODE REVIEW

File: java-engine/src/main/java/io/getunleash/engine/UnleashEngine.java [hunk 1]
```diff
@@ -66,21 +63,25 @@ public void countVariant(String flagName, String variantName) {
...
         try {
-            System.out.println(str); // TODO use a logging library. SLF4J?
-            return reader.forType(typeReference).readValue(str);
-        } catch (IOException e) {
-            throw new YggdrasilParseException(str, typeReference.getClass(), e);
+            String str = pointer.getString(0, UTF_8);
+            try {
+                return mapper.readValue(str, clazz);
+            } catch (IOException e) {
+                System.out.println("Failed to parse response from Yggdrasil: " + str);
```
Comment (sighphyre) [inline java-engine/src/main/java/io/getunleash/engine/UnleashEngine.java:80]: Can we trim this? Certainly for benchmarks, streaming to std out is costly. I removed this line when I ran the original benchmarks
```suggestion

```

Comment (gastonfournier) [reply java-engine/src/main/java/io/getunleash/engine/UnleashEngine.java:80]: Yes, I'd like to integrate with SLF4J and add this as debug log, but I'd like to keep it for now to later replace all System.out with the logger

Comment (sighphyre) [reply java-engine/src/main/java/io/getunleash/engine/UnleashEngine.java:80]: Fair enough, but you'll need to trim this out for your benchmark runs to get anything honest

**Review by sighphyre — APPROVED** [review]
G

File: java-engine/src/main/java/io/getunleash/engine/UnleashEngine.java (hunk 1 above)
Comment (gastonfournier) [reply java-engine/src/main/java/io/getunleash/engine/UnleashEngine.java:80]: Yes, the logger will just be disabled for benchmarks

**Pull request merged**